torch/fx/traceback.py

```diff
@@ -14,6 +14,31 @@
 @compatibility(is_backward_compatible=False)
 @contextmanager
 def preserve_node_meta():
+    """
+    Preserves state of ``meta`` field of a graph node
+
+    This function preserves a node's meta fields (stack_trace, nn_module_stack, val, tensor_meta, ...)
+    during fx.Transformer and aot_autograd transformations.
+
+    Example:
+
+    >>> import torch
+
+    >>> class SimpleAdd(torch.nn.Module):
+    >>>     def __init__(self):
+    >>>         super().__init__()
+    >>>     def forward(self, x, y):
+    >>>         return x + y
+
+    >>> with torch.fx.traceback.preserve_node_meta():
+    >>>     model = SimpleAdd()
+    >>>     fxg = torch.export.export(model, (torch.rand([1,5]), torch.rand([1,5])))
+    >>>     for n in fxg.graph.nodes:
+    >>>         print(n, n.meta)
+    >>>     fxg_new = fxg.run_decompositions()
+    >>>     for n in fxg_new.graph.nodes:
+    >>>         print(n, n.meta)
```

Comment:
Can someone from export check on this? I'm not sure I understand this example. Is `preserve_node_meta` supposed to affect `run_decompositions` specifically in export?

Reply:
@ezyang @angelayi Thanks for pointing this out. I made a new PR here https://github.com/pytorch/pytorch/pull/128681